src/libaktualizr/package_manager/androidmanager.cc

```diff
@@ -0,0 +1,105 @@
+#include <regex>
+
+#include "androidmanager.h"
+
+#include "utilities/utils.h"
+
+Json::Value AndroidManager::getInstalledPackages() const {
+  Json::Value packages(Json::arrayValue);
+  Json::Value package;
+  package["name"] = "aktualizr";
+  package["version"] = "n/a";
+  packages.append(package);
+  return packages;
+}
+
+Uptane::Target AndroidManager::getCurrent() const {
+  std::vector<Uptane::Target> installed_versions;
```

Comment:
Looks like there is a lot going on behind the scene between this and the aktualizr-android code. I don't object merging soon since it's platform-specific code but it would be nice to have quick summaries:

- what are exactly the roles of the different components and how they interact with the state machine?
- do you think the need to structure the code like that came from limitations of our current package manager api (not flexible enough)? If so, do you have suggestions to change it to handle this scenario more nicely?

Reply:
Existing package manager interface doesn't fit well while aktualizr is in role of a slave within aktualizr-android application. Extra entity (dispatcher) that I've introduced is required in that case as a sync. point between the sides. I did so to minimize interaction complexity and decrease amount of data to be exposed. And it brings me faster to the point I could play with an update mechanism on Android side.  

However I have the same doubts about clearness of the current solution. I'll look what I can do to simplify it.

While entire update approach on Android is not well-established yet, I have no precise suggestions about changing package manager api.

Reply:
Thanks for your summary!

If you don't manage to simplify the integration dramatically, I think it could wait. But some documentation about the complete installation process with details about each actors' interactions would be greatly appreciated (when you think it's ready). Either an external doc or helpful code comments would work.